crates/codegen/runtime/generator/src/parser/codegen/keyword_scanner_definition.rs

```diff
@@ -82,6 +83,54 @@ impl KeywordScannerDefinitionCodegen for KeywordScannerDefinitionRef {
 impl KeywordScannerDefinitionCodegen for model::KeywordValue {
     fn to_scanner_code(&self) -> TokenStream {
         // This is a subset; let's reuse that
-        ScannerDefinitionNode::from(self.clone()).to_scanner_code()
+        model::Scanner::from(self.clone()).to_scanner_code()
+    }
+}
+
+/// A newtype wrapper around [`model::KeywordItem`] that only has a single atom value.
+///
+/// The main usage for this type is to construct a keyword trie, as trie will
+/// only work with single atom values and keyword promotion needs to additionally account for
+/// keyword reservation, rather than just literal presence.
+#[derive(Clone)]
+pub struct KeywordScannerAtomic(Rc<model::KeywordItem>);
```

Comment:
This name doesn't describe very well IMO. Atomic virtually always has to do with concurrency.

Reply:
Agreed, the naming is unfortunate but I wanted to retain a mental link between this and model::KeywordValue::Atom`. Do you have any suggestions how we might keep the link but not use the loaded `Atomic`?

Reply:
Renamed KeywordScannerAtomic to KeywordItemAtom in https://github.com/NomicFoundation/slang/pull/1003/commits/9d903eceaaa6c85fe44fd8151fa23cebf21d3a50